lib/widgets/drive/drive-scanner.coffee

```diff
@@ -0,0 +1,99 @@
+###
+The MIT License
+
+Copyright (c) 2015 Resin.io, Inc. https://resin.io.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+###
+
+EventEmitter = require('events')
+_ = require('lodash')
+compare = require('./compare')
+
+module.exports = class DriveScanner extends EventEmitter
+
+	###*
+	# @summary Dynamically detect changes in drive list
+	# @name DriveScanner
+	# @class
+	# @public
+	# @memberof visuals
+	#
+	# @param {Function} driveFinder - drivelist
+	# @param {Object[]} [options] - scan options
+	# @param {Number} [options.interval] - interval
+	# @param {Object[]} [options.drives] - current drives
+	#
+	# @example
+	# scanner = new DriveScanner driveFinder,
+	#		interval: 1000
+	#		drives: [
+	#			{ foo: 'bar' }
+	#		]
+	###
+	constructor: (driveFinder, options = {}) ->
+		super()
+		_.defaults(options, interval: 1000, drives: [])
```

Comment:
Also, the whole constructor could have some empty lines in between, otherwise it looks a little bit too compacted for the eye:

```coffee
constructor: (driveFinder, options = {}) ->
   super()

   _.defaults options,
      interval: 1000
      drives: []

   @drives = options.drives

   @interval = setInterval =>
      @scan(driveFinder)
   , options.interval

   @scan(driveFinder)
```